src/CodeGenerators/SettingsGen/Parser/SettingsFromAttributeParser.cs

```diff
@@ -0,0 +1,174 @@
+
+		/// <summary>
+		/// Property Attributes to look for
+		/// </summary>
+		public ISet<string> PropertyAttributes { get; private set; }
+
+		/// <summary>
+		/// List of section classes and the corresponding with their section name
+		/// </summary>
+		public IList<(string sectionName, INamedTypeSymbol className)> Classes { get; } = new List<(string, INamedTypeSymbol)>();
+
+		/// <summary>
+		/// Constructor
+		/// </summary>
+		internal SettingsFromAttributeParser(ISet<string> attributesToFind, IContextWrapper contextWrapper)
+		{
+
+			if (attributesToFind.Count == 0)
+			{
+				throw new ArgumentException("Attributes to find list has no elements");
+			}
+			m_contextWrapper = contextWrapper;
+			ClassAttributes = attributesToFind;
+			PropertyAttributes = new HashSet<string>()
+			{
+				AttributeNames.Ignore,
+				AttributeNames.Parameter,
+				AttributeNames.Required,
+			};
+		}
+
+		/// <inheritdoc />
+		public void OnVisitSyntaxNode(GeneratorSyntaxContext context)
+		{
+			(IList<IAttributeWrapper> attributes, INamedTypeSymbol? classSymbol) = m_contextWrapper.GetAttributes(context);
+
+			if (attributes.Count > 0 && classSymbol != null)
+			{
+				foreach (IAttributeWrapper attribute in attributes)
+				{
+					if (ClassAttributes.Any(x => string.Equals(attribute.Name, x, StringComparison.OrdinalIgnoreCase)))
+					{
+						// Get the name value of the class
+						string name = classSymbol.Name;
+
+						if (attribute.Arguments.TryGetValue(nameof(ParameterModel.Name), out string value))
+						{
+							name = value;
+						}
+
+						Classes.Add((name, classSymbol));
+					}
+				}
+			}
+		}
+
+		/// <inheritdoc/>
+		public SettingsXmlModel GetSettings()
+		{
+			SettingsXmlModel settings = new();
+
+			foreach ((string name, INamedTypeSymbol classSymbol) in Classes.OrderBy(x => x.sectionName))
+			{
+				SectionModel sectionModel = new()
+				{
+					Name = name
+				};
+
+				// Get properties for a given class
+				IEnumerable<IPropertySymbol> properties = classSymbol.GetMembers()
+					.Where(m => m.Kind == SymbolKind.Property).Cast<IPropertySymbol>();
+
+				foreach (IPropertySymbol property in properties)
+				{
+					IList<IAttributeWrapper>? attrs = m_contextWrapper.GetAttributes(property);
+					if (TryCreateParameter(property.Name, attrs, out ParameterModel parameterModel))
+					{
+						sectionModel.Parameters.Add(parameterModel);
+					}
+				}
+
+				settings.Sections.Add(sectionModel);
+			}
+
+			return settings;
+		}
+
+		private bool TryCreateParameter(string propertyName, IList<IAttributeWrapper> attributes, out ParameterModel parameterModel)
+		{
+			parameterModel = new ParameterModel();
+
+			// If it has the ignore attribute don't generate it
+			if (attributes.Any(x => string.Equals(x.Name, AttributeNames.Ignore, StringComparison.OrdinalIgnoreCase)))
+			{
+				return false;
+			}
+
+			// Handle parameter attribute present
+			IEnumerable<IAttributeWrapper> parameters = attributes.Where(
```

Comment:
Using `FirstOrDefault` and then checking for `null` could be better than `Where`/`ToList`/`Count`/`First`